File: kdevelop/languages/ruby/app_templates/dcopservice/dcopservice.rb

class %{APPNAME} < KDE::DCOPObject

    k_dcop 'QString string(int)',
           'QStringList list()',
           'void add(QString)',
           'bool remove(QString)',
           'bool exit()'

    def initialize() 
        super("serviceInterface")
        puts "Starting new service... "
        @list = []
    end
    
    def string(idx)
        return @list[idx]
    end
    
    def list()
        return @list
    end
    
    def add(arg)
        puts "Adding " + arg + " to the list"
        @list << arg
    end
    
    def remove(arg)
	    @list.delete(arg) ? true : false
    end
    
    def exit()
        $kapp.quit()
        return true
    end
end

File: languages/cpp/app_templates/dcopservice/dcopservice.cpp
%{CPP_TEMPLATE}

#include "%{APPNAMELC}.h"
#include <kdebug.h>
#include <kapplication.h>

%{APPNAME}::%{APPNAME}() : DCOPObject("serviceInterface")
{
	kdDebug() << "Starting new service... " << endl;
	m_List = QStringList();
}

%{APPNAME}::~%{APPNAME}()
{
	kdDebug() << "Going away... " << endl;
}

QString %{APPNAME}::string(int idx)
{
	return *m_List.at(idx);
}

QStringList %{APPNAME}::list()
{
	return m_List;
}

void %{APPNAME}::add(QString arg)
{
	kdDebug() << "Adding " << arg << " to the list" << endl;
	m_List << arg;
}

bool %{APPNAME}::remove(QString arg)
{
	QStringList::Iterator it = m_List.find(arg);
	if (it != m_List.end())
	{
		m_List.remove(it);
	}
	else
		return false;
	return true;
}

bool %{APPNAME}::exit()
{
	kapp->quit();
}
